R/epidist.R

```diff
@@ -352,21 +353,28 @@ validate_epidist <- function(epidist) {
       ) %in%
         attributes(epidist)$names,
     "epidist must contain a disease (single character string)" =
-      is.character(epidist$disease$disease) &&
-        length(epidist$disease$disease) == 1,
+    checkmate::test_character(epidist$disease$disease, len = 1),
     "epidist must contain an epidemiological distribution" =
-      is.character(epidist$epi_dist) && length(epidist$epi_dist) == 1,
+      checkmate::test_character(epidist$epi_dist, len = 1),
     "epidist must contain a <distribution> or <distcrete> distribution or NA" =
-      inherits(epidist$prob_dist, c("distribution", "distcrete")) ||
-        is.na(epidist$prob_dist) || is.character(epidist$prob_dist),
+    checkmate::test_multi_class(
+      epidist$prob_dist, classes = c("distribution", "distcrete")
+    ) || checkmate::test_character(
+      epidist$prob_dist, any.missing = TRUE, len = 1
+    ),
     "epidisit must contain uncertainty, summary stats and metadata" =
       all(
         is.list(epidist$uncertainty),
         is.list(epidist$summary_stats),
         is.list(epidist$metadata)
       ),
     "epidist must contain a citation" =
-      inherits(epidist$citation, "bibentry")
+      inherits(epidist$citation, "bibentry"),
+    "epidist distribution must be either <distribution> or <distcrete>" =
+      inherits(epidist$prob_dist, c("distribution", "distcrete")) ||
+        is.na(epidist$prob_dist) || is.character(epidist$prob_dist),
```

Comment:
Possibly a good candidate for `test_multi_class()` - why is `NA` allowed here again?

Reply:
This is a duplicated check, it is now removed: fb6503ce1e1b15e5681fe17909db56cf245ede06.